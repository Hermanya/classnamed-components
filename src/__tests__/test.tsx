import * as React from 'react'
import {create} from 'react-test-renderer';
import {classnamed} from '..'

describe('classnamed', () => {
  it('renders', () => {
    const MySection = classnamed("section")`px-6 py-4 bg-white`;
    const component = create(
      <MySection></MySection>
    ).toJSON();
    expect(component).toMatchSnapshot()
  })

  it('renders with an additional className', () => {
    const MySection = classnamed("section")`px-6 py-4 bg-white`;
    const component = create(
      <MySection className="text-dark"></MySection>
    ).toJSON();
    expect(component).toMatchSnapshot()
  })

  it('renders with an additional className', () => {
    type myButtonProps = {
      size?: 'big' | 'small'
    }
    const MyButton = classnamed('button')`
      btn shadow
      ${(props: myButtonProps) => ({
        big: 'btn-lg',
        small: 'btn-sm',
      }[props.size || ''] || '')}
    ` as React.SFC<myButtonProps>

    expect(create(
      <MyButton size="small">small</MyButton>
    ).toJSON()).toMatchSnapshot();

    expect(create(
      <MyButton>regular</MyButton>
    ).toJSON()).toMatchSnapshot();

    expect(create(
      <MyButton as="a">link</MyButton>
    ).toJSON()).toMatchSnapshot();
  })

  it('renders with no initial className', () => {
    const MySection = classnamed("section")``;
    const component = create(
      <MySection></MySection>
    ).toJSON();
    expect(component).toMatchSnapshot()
  })
})
