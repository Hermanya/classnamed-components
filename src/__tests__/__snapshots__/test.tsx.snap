// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classnamed renders 1`] = `
<section
  className="px-6 py-4 bg-white"
/>
`;

exports[`classnamed renders with an additional className 1`] = `
<section
  className="px-6 py-4 bg-white text-dark"
/>
`;

exports[`classnamed renders with an additional className 2`] = `
<button
  className="btn shadow btn-sm"
  size="small"
>
  small
</button>
`;

exports[`classnamed renders with an additional className 3`] = `
<button
  className="btn shadow"
>
  regular
</button>
`;

exports[`classnamed renders with an additional className 4`] = `
<a
  className="btn shadow"
>
  link
</a>
`;

exports[`classnamed renders with no initial className 1`] = `
<section
  className=""
/>
`;
